src/core/Akka.Streams/CodeGen/FanInShape.tt

```diff
@@ -75,11 +75,13 @@ namespace Akka.Streams
         /// TBD
         /// </summary>
         /// <param name="n">TBD</param>
-        /// <exception cref="ArgumentException">TBD</exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// This exception is thrown when the specified <paramref name="n" /> is less than or equal to zero.
+        /// </exception>
         /// <returns>TBD</returns>
         public Inlet<T1> In(int n)
         {
-            if (n <= 0) throw new ArgumentException("n must be > 0", "n");
+            if (n <= 0) throw new ArgumentOutOfRangeException(nameof(n), "n must be > 0");
```

Comment:
I'll revert those, but

* No one catches/processes Argument* type exceptions.
* None of our tests fail because of the change.
* It's the correct exception to throw.